fix: added 'autoname' to reserved keywords

raizasafeel opened the pull request:
Added autoname to reserved keywords to prevent breaking when used as a fieldname.

Before:
![ScreencastFrom2025-12-2614-55-14-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/1cb97765-0f58-42b5-8a8d-c4e56f34cac0)

After:
<img width="2404" height="1069" alt="Screenshot From 2s025-12-26 14-58-11" src="https://github.com/user-attachments/assets/b8bccdf5-96df-43b0-adce-5a4c1a40b910" />

raizasafeel committed:
fix: added 'autoname' to reserved keywords

iamejaaz commented:
@ankush It does not allow the autoname value to be updated because autoname is included in the reserved keywords, which is why updating the doctype fails.
https://github.com/frappe/frappe/blob/db4e83483799f1c546341a804252fbcf94371e54/frappe/model/base_document.py#L277

akhilnarang commented:
Can have some different handling for this, not part of reserved keywords.

Something where we check `keyword.kwlist` and few other special names like this, and prevent using them in a doctype.

Commit:
fix: prevent using autoname as DocField name

- Add validation for autoname in check_invalid_fieldnames()
- Treat Python keywords as reserved fieldnames

Commit:
fix: move autoname out of reserved keywords

Commit:
fix: move python keywords out of reserved keywords

ankush commented:
@raizasafeel setup git to avoid this: 
<img width="773" height="232" alt="image" src="https://github.com/user-attachments/assets/b0afd039-f0f2-4c9c-8aea-7a28b882cd50" />


https://gameplan.frappe.cloud/g/space/94/discussion/1930/git-gud-for-using-git-at-frappe

ankush commented on frappe/core/doctype/doctype/doctype.py:
```diff
@@ -1278,6 +1278,8 @@ def check_illegal_characters(fieldname):
 		validate_column_name(fieldname)
 
 	def check_invalid_fieldnames(docname, fieldname):
+		INVALID_DOCFIELD_NAMES = frozenset(("autoname",))
```
Add an exception for `doctype="DocType"` 😄

raizasafeel replied:
got it! thanks!

Commit:
fix: skip reserved docfield name check for 'DocType'

iamejaaz commented on frappe/model/base_document.py:
```diff
@@ -2,6 +2,7 @@
 # License: MIT. See LICENSE
 import datetime
 import json
+import keyword
```
@raizasafeel 
Can this be removed, since it is not used anywhere?

raizasafeel replied:
yes, sorry about that
